pkg/storage/stores/shipper/compactor/compactor.go

```diff
@@ -87,6 +88,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.DeleteRequestCancelPeriod, "boltdb.shipper.compactor.delete-request-cancel-period", 24*time.Hour, "Allow cancellation of delete request until duration after they are created. Data would be deleted only after delete requests have been older than this duration. Ideally this should be set to at least 24h.")
 	f.IntVar(&cfg.MaxCompactionParallelism, "boltdb.shipper.compactor.max-compaction-parallelism", 1, "Maximum number of tables to compact in parallel. While increasing this value, please make sure compactor has enough disk space allocated to be able to store and compact as many tables.")
 	f.StringVar(&cfg.DeletionMode, "boltdb.shipper.compactor.deletion-mode", "whole-stream-deletion", fmt.Sprintf("(Experimental) Deletion mode. Can be one of %v", strings.Join(deletion.AllModes(), "|")))
+	f.StringVar(&cfg.Address, "boltdb.shipper.compactor.address", "", "host and port where the compactor API is listening")
```

Comment:
Adding this config to the compactor looks out of place to me. New users already get confused by our configs, so it is better to duplicate it a little than add to the confusion. It could be just me. What do you think?

Reply:
I went back and forth on this. If we leave it off the compactor config, it becomes a required property on:
- frontend
- querier
- ruler

Do you think that's clearer for the user than having the compactor know it's own address?

Reply:
is there also a case to be made for the `common` config, since it's common across multiple components?

Reply:
Common Config says `// Values defined under this common configuration are supersede if a more specific value is defined.`

Does that imply that there's also some more specific override elsewhere? If not, it would be a fine place for this to go.

Reply:
I don't think that would apply in this case, if we moved it to the common config so that was the only place it's defined, then there would be no superseding. What that comment is meant to convey is that after things from the common config (like storage or ring configs) are applied to the many places those are used, the `config.yaml` is read again, so if there is say, a specific ring config for the scheduler, that will supersede whatever was put in common.

Reply:
yeah, I thought about using `common` config but due to that comment I thought it was not the right place, same as Travis. If we can use it for this use case then it would be a better option.